### PR TITLE
cli: move the demo cluster to new sub-package `democluster` (5/)

### PR DESCRIPTION
**Pull request by knz**
First 15 commits are from #67459 and prior.
Please ignore those (they will disappear once the prefix PR is merged)

Only the last 5 commits belong to this PR.

This is part of a family of PRs that aim to encapsulate CLI functionality into finer grained packages.

### CODE REVIEW

**Review by otan — APPROVED** [review]
LGTM, one question about os.Stdout

File: pkg/cli/auth.go [hunk 1]
```diff
@@ -175,7 +175,7 @@ func runLogout(cmd *cobra.Command, args []string) (resErr error) {
             id AS "session ID",
             "revokedAt" AS "revoked"`,
 		username)
-	return sqlExecCtx.RunQueryAndFormatResults(sqlConn, os.Stdout, logoutQuery)
+	return sqlExecCtx.RunQueryAndFormatResults(sqlConn, os.Stdout, stderr, logoutQuery)
```
[the remark this answers is not in the document]
Comment (knz) [reply pkg/cli/auth.go:178]: The commit message was about removing changes to the standard streams in the sub-packages. The `cli` package remains very much responsible for "plugging things together" and so gets to choose the streams.

**Comment by knz** [comment]
TFYR!

bors r=otan

**Comment by knz** [comment]
bors r=otan

**Comment by knz** [comment]
the bors run is currently flaking on #64866. I hope it gets retried.

**Pull request merged**